assets/src/main/kotlin/ktx/assets/assets.kt

```diff
@@ -212,3 +215,99 @@ inline fun <reified Type : Any, Parameters : AssetLoaderParameters<Type>> AssetM
     suffix: String? = null) {
   setLoader(Type::class.java, suffix, assetLoader)
 }
+
+/**
+ * A class for managing a group of assets together, such that they can be loaded or unloaded en masse.
+ * [AssetGroup] should be subclassed and the assets should be defined by properties in the subclass by using [asset] or,
+ * [delayedAsset], which can be used as delegates. If not using delegates, it is possible to selectively load and
+ * unload them, in which case it is not advised to use [loadAll] and [unloadAll], because AssetManager's reference
+ * counts may be thrown off.
+ *
+ * If fine control over individual assets of this group is not needed, a strategy to avoid the complexity of
+ * AssetManager's reference counting is to:
+ * - Use [asset]/[delayedAsset] exclusively as delegates.
+ * - Do not mix and match use of [asset] and [delayedAsset].
+ * - Never access any of the asset properties until the group is finished loading.
+ * - Do not initially call [loadAll] if using [asset] (as opposed to [delayedAsset]).
+ * @param manager The [AssetManager] that will handle loading and unloading of this group.
+ * @param filePrefix A string that will be prefixed to any file path parameter passed to [asset] or [delayedAsset].
+ */
+abstract class AssetGroup(val manager: AssetManager, protected val filePrefix: String = "") {
+  /** The backing set containing the assets of this group. There is no need to manually add assets to this set if using
+   * [asset] or [delayedAsset]. */
+  protected val members = ObjectSet<Asset<*>>()
+
+  /** Queues all assets of this group for loading by the associated [AssetManager].
+   * If any assets of this group are already loaded, their load counts in the AssetManager will still be incremented. */
+  fun loadAll() {
+    for (member in members)
+      member.load()
+  }
+
+  /** Unloads all of the assets in this group. If any of the assets are dependencies of assets outside this group, or if
+   * they were loaded more than once, their load counts will only be decremented by one and the associated [AssetManager]
+   * will retain them. */
+  fun unloadAll() {
+    for (member in members)
+      member.unload()
+  }
+
+  /** Unloads all of the assets in this group, catching any exceptions. If any of the assets are dependencies of
+   * assets outside this group, or if they were loaded more than once, their load counts will only be decremented by one
+   * and the associated [AssetManager] will retain them.
+   * @param onError Called in response to each caught exception. By default, the exceptions are ignored.
+   * */
+  inline fun unloadAllSafely(onError: (Asset<*>, Exception) -> Unit = { _, exception -> exception.ignore() }) {
+    for (member in `access$members`){
```

Comment:
That's an unusual syntax. For the sake of clarity, this function does not have to be inlined - it's unlikely to be performance critical either way.

Reply:
I was inlining this more so to allow the response to an error be to return early. But I suppose that would be a very rare case. You would either be using this to debug, or to unload everything regardless of error.

Reply:
Not sure if you'd like to or should break early here, to be honest.

Reply:
I suppose if someone wanted to break early, they would use wrap an `unloadAll()` call in a try/catch instead of using `unloadAllSafely()`.